feat: upstream spec to ecotone (#8)
kroma-network/kroma-specs: specs/protocol/exec-engine.md

  `uint256` range.

- zeroes and ones are the count of zero and non-zero bytes respectively in the **full** encoded
  signed transaction.

- `l1BaseFee` is the L1 base fee of the latest L1 origin registered in the L2 chain.

- `l1BlobBaseFee` is the blob gas price, computed as described in [EIP-4844][4844-gas] from the
  header of the latest registered L1 origin block.

Conceptually what the above function captures is the formula below, where `compressedTxSize =
(zeroes*4 + ones*16) / 16` can be thought of as a rough approximation of how many bytes the
transaction occupies in a compressed batch.

`(compressedTxSize) * (16*l1BaseFee*lBaseFeeScalar + l1BlobBaseFee*l1BlobBaseFeeScalar) / 1e6`

The precise cost function used by Ecotone at the top of this section preserves precision under
integer arithmetic by postponing the inner division by 16 until the very end.

[4844-gas]: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4844.md#gas-accounting

The two base fee values and their respective scalars can be accessed in two interchangeable ways:

- read from the deposited L1 attributes (`l1BaseFeeScalar`, `l1BlobBaseFeeScalar`, `basefee`,
  `blobBaseFee`) of the current L2 block

> - read from the L1 Block Info contract (`0x4200000000000000000000000000000000000015`)

Reviewer: ```suggestion
- read from the L1 Block Info contract (`0x4200000000000000000000000000000000000002`)
```

  - using the respective solidity getter functions
  - using direct storage-reads:
    - basefee `uint256` in slot `1`

>     - blobBaseFee `uint256` in slot `7`

Reviewer: ```suggestion
    - blobBaseFee `uint256` in slot `8`
```

2. The rollup node maintains latest head from engine (poll `eth_getBlockByNumber` and/or maintain a header subscription)
3. The rollup node activates sync if the engine is out of sync but not syncing through P2P (`eth_syncing`)
4. The rollup node inserts blocks, derived from L1, one by one, potentially adapting to L1 reorg(s),
   as outlined in the [rollup node spec].

[rollup node spec]: rollup-node.md

## Ecotone: disable Blob-transactions

[EIP-4844] introduces Blob transactions: featuring all the functionality of an [EIP-1559] transaction,
plus a list of "blobs": "Binary Large Object", i.e. a dedicated data type for serving Data-Availability as base-layer.

With the Ecotone upgrade, all Cancun L1 execution features are enabled, with [EIP-4844] as exception:

> as a L2, the OP-Stack does not serve blobs, and thus disables this new transaction type.

Reviewer 2: ```suggestion
as a L2, the Kroma-network does not serve blobs, and thus disables this new transaction type.
```
